src/Illuminate/Database/Connectors/ConnectionFactory.php

```diff
@@ -98,6 +98,13 @@ protected function getReadConfig(array $config)
     {
         $readConfig = $this->getReadWriteConfig($config, 'read');
 
+        // Check if we have multiple slave hosts
```

Comment:
Sorry, didn't get that. Are you talking about this modification not adding any value ?

Motivation: 1 master - X slaves. At run time, pick a slave at random for all read queries. Very common scenario. Can do the same is config/databases.php , but not sure how php artisan config:cache will behave then.